### DISCUSSION TITLE
cloud firestore error, uses unsafe or unchecked operations.

### DISCUSSION THREAD

**Discussion by queenmellow** [firebase/flutterfire#10756] [Q&A]
i added cloud firestore to my app and i got this error 

`Note: C:\Users\Benaiah\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.0\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestoreMessageCodec.java uses unchecked or unsafe operations.`

can someone help with what to do. plz

**Comment by lemohammed** [comment]
Same I am getting 

```
.pub-cache/hosted/pub.dev/cloud_firestore-4.5.0/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestoreMessageCodec.java:412: error: where(Filter) is not public in Query; cannot be accessed from outside package
        query = query.where(filter);
```

**Comment by s4nk37** [comment]
same

**Comment by WangShuan** [comment]
Same,

I got this:
Note: /Users/wangxuan/.pub-cache/hosted/pub.dev/cloud_firestore-4.5.0/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestoreMessageCodec.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/wangxuan/.pub-cache/hosted/pub.dev/firebase_messaging-14.4.0/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

**Comment by sahil1213-ux** [comment]
i am getting this can please someone help me:

Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\sahil\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.9.1\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestoreMessageCodec.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

**Comment by osaho** [comment]
I am also getting the same problem and don't know what to do

Launching lib\main.dart on Infinix X650 in debug mode...
main.dart:1
Note: C:\Users\DELL\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.9.1\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestoreMessageCodec.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
